src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -618,25 +630,43 @@ private void ExecuteClusterTransactionOneByOne(List<ClusterTransactionCommand.Si
             try
             {
                 var index = command.Index;
+
                 var options = mergedCommands.Options[index];
                 if (exception == null)
                 {
-                    Task indexTask = null;
                     if (options.WaitForIndexesTimeout != null)
                     {
-                        indexTask = BatchHandler.WaitForIndexesAsync(DocumentsStorage.ContextPool, this, options.WaitForIndexesTimeout.Value,
+                        var indexTask = BatchHandler.WaitForIndexesAsync(DocumentsStorage.ContextPool, this, options.WaitForIndexesTimeout.Value,
                             options.SpecifiedIndexesQueryString, options.WaitForIndexThrow,
                             mergedCommands.LastDocumentEtag, mergedCommands.LastTombstoneEtag, mergedCommands.ModifiedCollections);
-                    }
 
-                    var result = new BatchHandler.ClusterTransactionCompletionResult
+                        var removeTask = ClusterTransactionWaiter.CreateTask(command.Options.TaskId, command.Index, out _);
```

Comment:
don't we need to always create this? regardless whether we have indexes or not?

Reply:
but in the previous commit this was _always_ created.

Reply:
the remove task that we are creating here is relevant only for a failover with wait for index